src/mprisServer.c

```diff
@@ -75,6 +85,85 @@ static const char xmlForNode[] =
 static GDBusConnection *globalConnection = NULL;
 static GMainLoop *loop;
 
+static gboolean bytecodeCompiled;
+
+static GVariant* produceScalarString(const char *valueStr) {
+	return g_variant_new_string(valueStr);
+}
+
+static GVariant* produceSingleStringArray(const char *valueStr) {
+	GVariantBuilder arrayBuilder;
+	g_variant_builder_init(&arrayBuilder, G_VARIANT_TYPE("as"));
+
+	g_variant_builder_add(&arrayBuilder, "s", valueStr);
+
+	return g_variant_builder_end(&arrayBuilder);
+}
+
+static GVariant* produceScalarInt(const char *valueStr) {
+	gint32 value = atoi(valueStr);
+
+	if (value <= 0) {
+		return NULL;
+	}
+
+	return g_variant_new_int32(value);
+}
+
+static char* chompCr(char *str) {
+	size_t len = strlen(str);
+
+	if (len > 0 && str[len - 1] == '\r') {
+		str[len - 1] = '\0';
+	}
+
+	return str;
+}
+
+static GVariant* produceArrayOfTokens(const char *valueStr) {
+	char **tokens = g_strsplit(valueStr, "\n", 16);	// 16 genre records should be enough for anybody
```

Comment:
@kravich Why not just use -1 for max_tokens?

Do you see a potential problem here?

Reply:
Sorry for the late reply.

Originally I thought that limiting tokens up to some fixed count is a good protection against malicious input with a lot of "\n" that will cause program to allocate enormous amount of dynamic strings with corresponding problems. But now it seems for me that not having limitation for the max count of genres is better than protecting from very very unlikely case. So, I changed max_tokens to -1.